src/Appwrite/Platform/Tasks/Restore.php

```diff
@@ -0,0 +1,300 @@
+<?php
+
+namespace Appwrite\Platform\Tasks;
+
+use Exception;
+use Utopia\App;
+use Utopia\DSN\DSN;
+use Utopia\Platform\Action;
+use Utopia\CLI\Console;
+use Utopia\Storage\Device;
+use Utopia\Storage\Device\DOSpaces;
+use Utopia\Storage\Device\Local;
+use Utopia\Validator\Boolean;
+use Utopia\Validator\Text;
+
+class Restore extends Action
+{
+    public const BACKUPS_PATH = '/backups';
+    public const DATADIR = '/var/lib/mysql';
+    protected ?DOSpaces $s3 = null;
+    protected string $xtrabackupContainerId;
+    protected int $processors = 1;
+
+    public function __construct()
+    {
+        $this
+            ->desc('Restore a DB')
+            ->param('id', '', new Text(19), 'The backup identification')
+            ->param('cloud', null, new Boolean(true), 'Download backup from cloud or use local directory')
+            ->param('database', null, new Text(10), 'The Database name for example db_fra1_01')
+            ->callback(fn ($id, $cloud, $project) => $this->action($id, $cloud, $project));
+    }
+
+    public static function getName(): string
+    {
+        return 'restore';
+    }
+
+    public function action(string $id, string $cloud, string $database): void
+    {
+        $this->checkEnvVariables();
+        $this->setContainerId();
+        $this->setProcessors();
+
+        $datadir = self::DATADIR;
+
+        try {
+            $dsn = new DSN(App::getEnv('_APP_CONNECTIONS_BACKUPS_STORAGE', ''));
+            $this->s3 = new DOSpaces('/' . $database . '/full', $dsn->getUser(), $dsn->getPassword(), $dsn->getPath(), $dsn->getParam('region'));
+            $this->s3->setTransferChunkSize(40 * 1024 * 1024); // 5MB
+        } catch (\Exception $e) {
+            Console::error($e->getMessage() . 'Invalid DSN.');
+            Console::exit();
```

Comment:
Let's throw instead of error/exit, for all cases in this task

Reply:
Cool!